client/protocols/analytics.py

```diff
@@ -106,13 +110,14 @@ def log_run(self, messages):
             history['question'] = questions
 
         # Update attempt and correctness counts for the graded questions
-        for question in questions:
+        for question in history['question']:
             detail = history['questions']
```

Comment:
`question` is the currently requested quetion(s). When I first made this, I thought that there would only be one question - but it turned out to be a list of questions - though usually there are only one. I agree that this confusing - but I'd like the current naming schema for the protocol to preserve all of our existing analysis scripts. 

As for the for loop - It'd be a good idea to have a better name that is actually more representative of what it is. I'll push that up in a bit. 

`questions` contains the historical information for all questions.